flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/KeyedSerializationSchemaWrapper.java

```diff
@@ -37,6 +37,10 @@ public KeyedSerializationSchemaWrapper(SerializationSchema<T> serializationSchem
 		this.serializationSchema = serializationSchema;
 	}
 
+	public SerializationSchema<T> getSerializationSchema() {
```

Comment:
Why don't we forward the `open()` call but need to do casts when invoking?

Reply:
I did not want to add new features to a deprecated interface. The KeyedSerializationSchema is deprecated and that's why I did not want to add the `open()` method there. Therefore there is no method where I can forward it. Still I wanted to support the open for the top-level `SerializationSchema`, which is wrapped in that deprecated class.

Reply:
Ok, that seems fine. Good to merge then.